ui/lib/utils/tableColumnFactory.tsx

```diff
@@ -0,0 +1,256 @@
+import { Tooltip } from 'antd'
+import { max as _max } from 'lodash'
+import {
+  IColumn,
+  ColumnActionsMode,
+} from 'office-ui-fabric-react/lib/DetailsList'
+import React from 'react'
+import { getValueFormat } from '@baurine/grafana-value-formats'
+
+import {
+  Bar,
+  Pre,
+  TextWithInfo,
+  TextWrap,
+  DateTime,
+  HighlightSQL,
+  IColumnKeys,
+} from '@lib/components'
+
+type Bar = { [key: string]: string }
+type BarsConfig = {
+  displayTransKey?: string // it is same as avg field name default
+  bars: [Bar, Bar, Bar?] // [avg, max, min?]
+}
+
+export type IExtendColumn = IColumn & {
+  refFields?: string[]
+}
+
+function capitalize(s: string) {
+  return s.charAt(0).toUpperCase() + s.slice(1)
+}
+
+export function formatVal(val: number, unit: string) {
+  const formatFn = getValueFormat(unit)
+  return unit === 'short' ? formatFn(val, 0, 1) : formatFn(val, 1)
```

Comment:
This is too magic, I prefer to expose the logic to the caller. Not in all cases we want zero decimals for short values. For example, for averages, short value with decimals looks to be fine.

Reply:
`formatFn(val, 0, 1)` for short means if the value less than 1000, than we use zero decimals, else we use 1 decimal. This behavior is the exact same as before, I didn't change it. Currently, there are no averages that are short type.